### PR TITLE
cli: Use program length for deployments instead of 2x length

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

The default behavior for program deployments uses 2x the length of the deployed program. This behavior accommodated larger program deployments over time, but requires a greater upfront cost. With `solana program extend`, however, there's no need to lock up more SOL than necessary for a program.

#### Summary of Changes

Use the program length when deploying instead of 2x, and update the documentation to reflect that change, including how to extend a program.

There might be some overlap of changes by including the `extend` documentation, so let me know if you prefer that in a separate PR!

Fixes #

### CODE REVIEW

**Comment by pgarg66** [comment]
Looks good to me!

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,9 @@ Release channels have their own copy of this changelog:
   * Bigtable upload now includes entry summary data for each slot, stored in a
     new `entries` table
   * Forbid multiple values for the `--signer` CLI flag, forcing users to specify multiple occurrences of `--signer`, one for each signature
+  * New program deployments default to the exact size of a program, instead of
+    double the size. Programs must be extended with `solana program extend` to
+    accommodate a larger program before upgrading.
```
Comment (CriesofCarrots) [inline CHANGELOG.md:34]:
```suggestion
    double the size. Program accounts must be extended with `solana program extend`
    before an upgrade if they need to accommodate larger programs.
```

**Review by CriesofCarrots — COMMENTED** [review]
r+ nit in changelog text

File: CHANGELOG.md (hunk 1 above)
Comment (joncinque) [reply CHANGELOG.md]: Done!

**Review by CriesofCarrots — APPROVED** [review]

**Pull request merged**